packages/@sanity/preview/src/components/ObserveForPreview.js

```diff
@@ -44,7 +43,7 @@ const connect = props$ => {
   const isActive$ = sharedProps$.pipe(map(props => props.isActive !== false))
 
   return sharedProps$.pipe(
-    distinctUntilChanged((props, nextProps) => getId(props.value) === getId(nextProps.value)),
+    distinctUntilChanged((props, nextProps) => shallowEquals(props.value, nextProps.value)),
```

Comment:
How often does this compare? Only when directly targeted via patch, or more than that? I'm haven't seen the context.

Anyway, ``getId`` is broken so approved upfront.

Reply:
It does a shallow compare whenever the component gets passed new props. Buffered document can be one of many possible sources of this - why?